src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -383,7 +383,17 @@ private BoundExpression ConvertConcatExprToStringIfPossible(CSharpSyntaxNode syn
                         if (ConcatExprCanBeOptimizedWithToString(operand.Type))
                         {
                             var toString = GetSpecialTypeMethod(syntax, SpecialMember.System_Object__ToString);
```

Comment:
@TyOverby Yes, we shouldn't assume that some methods or types exist until we confirmed their existence.